Sources/UIKit/TextAlignmentConstraint.swift

```diff
@@ -0,0 +1,268 @@
+//
+//  TextAlignmentConstraint.swift
+//
+//  Created by Cameron Pulsford on 10/4/16.
+//  Copyright © 2016 Raizlabs. All rights reserved.
+//
+
+#if os(OSX)
+    import AppKit
+#else
+    import UIKit
+#endif
+
+private var TextAlignmentConstraintKVOContext = "BonMotTextAlignmentConstraintKVOContext" as NSString
+
+@objc(BONTextAlignmentConstraint)
+public class TextAlignmentConstraint: NSLayoutConstraint {
+
+    @objc(BONTextAlignmentConstraintAttribute)
+    public enum Attribute: Int, CustomStringConvertible {
+        case unspecified
+        case top
+        case capHeight
+        case xHeight
+        case firstBaseline
+        case lastBaseline
+        case bottom
+
+        var layoutAttribute: NSLayoutAttribute {
+            switch self {
+            case .top, .capHeight, .firstBaseline, .xHeight:
+                #if swift(>=3.0)
+                    return .top
+                #else
```

Comment:
You should add an extension on `NSLayoutAttribute` in `Compatibility.swift` and add:
```
extension NSLayoutAttribute {
   static var top = NSLayoutAttribute.Top
}
```
Ideally, once you've written the 3.0 code, getting 2.3 to work should be adding stuff to `Compatibility.swift`.